woodchipper/context.py

```diff
@@ -138,12 +139,22 @@ def _convert_to_loggable_value(value: Any) -> LoggableValue:
             )
 
 
+def _convert_to_valid_log_level(log_level: Union[str, int, Missing]) -> int:
```

Comment:
I don't love this as it feels too hand-hold-y. But failing fast is not really an option as the value can be updated at any point. This guaranties that any value passed to a logging context as a log level is made valid, so we do not throw when we get to the logging call.

Reply:
I like it.